src/convert.cc

```diff
@@ -0,0 +1,322 @@
+/*
+Copyright 2019 Linkworld Open Team
+                              YHTB
+*/
+#include "./frame.h"
+using uint = unsigned int;
+enum { Byte = 8, VintTotalBit = 64 };
+//----------------------------------------------------------------------------//
+// 各种对象到二进制的转换
+enum Type { FRS = 0x04, FSDB = 0x15 };
+
+// 获取64位无符号整形的第一二位
+inline uint8_t GetFlag(const vint num, const uint bits = 62) {
+  uint8_t tmp = num >> bits;
+  return reinterpret_cast<uint8_t>(tmp);
+}
+inline uint8_t GetFlag(const char *const bin, const uint bits = 62) {
+  uint8_t tmp = *bin >> bits;
+  return reinterpret_cast<uint8_t>(tmp);
+}
+//获取一个字段的长度 单位是byte
+inline uint GetLen(uint flag) {
+  switch (flag) {
+    case 0x0:
+      return 1;
+    case 0x1:
+      return 2;
+    case 0x2:
+      return 4;
+    case 0x3:
+      return 8;
+    default:
+      return 0;
+  }
+}
+// vint<=>bin
+char *vintTobin(const vint num) {
+  char *tmp;
+  switch (GetFlag(num)) {
+    case 0x0:
+      tmp = new char[1]{0};
+      break;
+    case 0x1:
+      tmp = new char[2]{0};
+      break;
+    case 0x2:
+      tmp = new char[4]{0};
+      break;
+    case 0x3:
+      tmp = new char[8]{0};
+      break;
+    default:
+      return nullptr;
+  }
+  *tmp |= num;
+  return tmp;
+}
+// 合并数据,将orginData 的数据合并到location 第四个参数的单位为byte
+void MergeData(const char OrginData[], char DirData[], uint *location,
```

Comment:
variable names must be lowercase with underscore.
Note that DirData name is wrong, use dest_data instead.
like:
```cpp
void MergeData(const char origin_data[], char dest_data[], uint *location, uint len)

```